internal/customize/customize.go

```diff
@@ -0,0 +1,310 @@
+	overlaySource, err := deployment.NewSrcFromURI(overlaysURI)
+	if err != nil {
+		return nil, fmt.Errorf("parsing overlay source URI %q: %w", overlaysURI, err)
+	}
+	d.OverlayTree = overlaySource
+
+	// Make sure that we define a valid installer config script if such was not defined
+	// during the build process of the ISO that is currently being customized.
+	if installerDep.Installer.CfgScript == "" {
+		autoInst, err := writeAutoInstaller(fs, string(output), mediaType)
+		if err != nil {
+			return nil, fmt.Errorf("writing default '%s' installer script: %w", autoInstallerScriptName, err)
+		}
+
+		d.Installer.CfgScript = autoInst
+	}
+
+	return d, nil
+}
+
+// prepareDeploymentPartitions produces a partition slice that is ready for
+// merge with a partition slice coming from the 'install.yaml' file that was
+// created during the `build-installer` command.
+//
+// Currently only additions of new partitions is supported. More complex logic
+// is not yet needed and needs to be defined once we flesh out the correct deployment
+// merging approach.
+func prepareDeploymentPartitions(src, add []*deployment.Partition) []*deployment.Partition {
+	preparedPartitionSlice := []*deployment.Partition{}
+
+	for i := range src {
+		if i == len(src)-1 {
+			// Assume the last partition is SYSTEM and mark its position in
+			// the new slice as nil. This is done in order to not break the
+			// deployment merge process and have either the SYSTEM partition
+			// not defined last, or have it mangled up with another partition.
+			preparedPartitionSlice = append(preparedPartitionSlice, nil)
+			continue
+		}
+
+		// Any existing partitions that are not the last partition are defined as
+		// empty partitions in the new slice. This will instruct the deployment merge
+		// logic to skip these entries and not merge them with the current deployment
+		// partitions.
+		preparedPartitionSlice = append(preparedPartitionSlice, &deployment.Partition{})
+	}
+
+	// Include the additional partitions to the newly produced slice
+	preparedPartitionSlice = append(preparedPartitionSlice, add...)
+
+	// Lastly, redefine the SYSTEM partition, as seen in src.
+	// This is done to ensure that the SYSTEM partition will always be defined
+	// last, as is the requirement for the OS.
+	// ---
+	// The final result will look like this:
+	// preparedPartitions = [{}, {}, nil, add..., SYSTEM].
+	//
+	// When this result is merged with the installer partition slice - '[EFI, RECOVERY, SYSTEM]';
+	// we will get the following merged result:
+	// mergedPartitions = [EFI, RECOVERY, add..., SYSTEM]
+	return append(preparedPartitionSlice, src[len(src)-1])
+}
+
+func resizeDisk(runner sys.Runner, disk string, diskSize install.DiskSize) error {
+	const defaultSize = "10G"
+
+	if diskSize == "" {
+		diskSize = defaultSize
+	} else if !diskSize.IsValid() {
+		return fmt.Errorf("invalid disk size definition '%s'", diskSize)
+	}
+
+	_, err := runner.Run("truncate", "-s", string(diskSize), disk)
```

Comment:
nit: I'd rather use os.File.Truncate which is then supporting the vfs interface and you could eventually handle the spare vs non spare case by writing some bytes at the end or not.